src/View/AbstractDirectoryView.vala

```diff
@@ -1977,7 +1989,36 @@ namespace Files {
             ));
             properties_menuitem.action_name = "common.properties";
 
-            if (get_selected_files () != null) {
+            Gtk.MenuItem? select_all_menuitem = null;
+            Gtk.MenuItem? unselect_all_menuitem = null;
+            Gtk.MenuItem? invert_selection_menuitem = null;
+            if (!all_selected) {
+                select_all_menuitem = new Gtk.MenuItem () {
+                    action_name = "common.select-all"
+                };
+                select_all_menuitem.add (new Granite.AccelLabel.from_action_name (
+                    _("Select All"),
+                    select_all_menuitem.action_name
+                ));
+            } else {
+                unselect_all_menuitem = new Gtk.MenuItem () {
+                    action_name = "common.select-all"
+                };
+                unselect_all_menuitem.add (new Granite.AccelLabel.from_action_name (
+                    _("Unselect All"),
```

Comment:
From my brief research, it seems "deselect" is the more common term in US English, at least. Not sure if you'd like to update the method/variable names as well, to match.

```suggestion
                    _("Deselect All"),
```

Reply:
OK, I assumed US English preferred "Unselect" as that is what is used for Gtk and other library functions.  Probably would have translated it to "Deselect" for UK English anyway.